app/src/main/java/org/thoughtcrime/securesms/audio/AudioRecorder.java

```diff
@@ -38,7 +36,7 @@ public AudioRecorder(@NonNull Context context) {
     this.context = context;
   }
 
-  public void startRecording() {
+  public void startRecording(InputBar inputBar) {
```

Comment:
Should we really add a dependency to InputBar in here? Should the AudioRecorder really know about InputBar?
I can see the code needs to wait for the end of that executor. Should the function instead take in a function, something like `onRecordingStarted` ? That way the caller can decide what to do without mingling the logic inside the AudioRecorder?

Reply:
Have modified to send a callback to `startRecording`